### PR TITLE
Update git submodules with gitee mirrors in PRC network

### PR DESCRIPTION
**Pull request by sammysun0711**
Git submodule update takes quite long time in PRC network, this PR aims to offer a optional method to update git submodules via gitee mirrors, which can efficiently reduce procedure runtime to improve UX for PRC developer.

### Details:
 - Add script to redirect git submodules to gitee mirrors and updates these recursively
 - OpenVINO wiki BuildingForLinux should be updated with instructions, please check patch as follow: [add_optional_submodule_update_with_gitee_in_wiki.txt](https://github.com/openvinotoolkit/openvino/files/7134508/add_optional_submodule_update_with_gitee_in_wiki.txt)

### Tickets:
 - XXX-52622

### CODE REVIEW

File: scripts/submodule_update_with_gitee.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+SCRIPT_SH=$(cd "$(dirname "${0}")" && pwd -P)/$(basename "${0}")
+
+[ -d "${REPO_DIR}" ] || die "${REPO_DIR} is not directory!"
+[ -f "${SCRIPT_SH}" ] || die "${SCRIPT_SH} does not exist!"
+
+pushd ${REPO_DIR} >/dev/null
+
+# Step 0: Check if .gitmodules file exsit
+[ -f ".gitmodules" ] || exit 0
```
Comment (ilya-lavrenov) [inline scripts/submodule_update_with_gitee.sh:57]: n.b. `exit 1` since script has not performed required actions?

Comment (sammysun0711) [reply scripts/submodule_update_with_gitee.sh:57]: `submodule_update_with_gitee.sh` will be called recursively to update submodules. If one submodule does not have `.gitmodules` file,  which means there is no sub-submodules, submodule update procedure for this repo will be skipped without error. It is desired behavior, hence `exit 0` is used. `exit 1` will stop bash script immediately if any submodules does not have sub-submodules.

**Comment by sammysun0711** [comment]
@ilya-lavrenov @azhogov please review it, Thanks!

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by sammysun0711** [comment]
@adk9 @SDxKeeper Could you please have a look at this PR? 
Besides, I need a maintainer to approve running workflow for ci/jenkins. Thanks!

**Pull request merged**